src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/ODataServiceDocumentResourceV2Provider.cs

```diff
@@ -119,7 +119,7 @@ public ODataServiceDocumentResourceV2Provider()
             }
 
             string serviceDocumentBaseAddress = null;
-            if (response.Status != HttpSourceResultStatus.NotFound)
```

Comment:
So this (and the case below in `RemoteV2FindPackageByIdResource`) are the only places that could possibly hit the `204 No Content`? For example, do any V3 endpoints go through this code path and have to check for `NoContent`?